addons/binding/org.openhab.binding.squeezebox/src/main/java/org/openhab/binding/squeezebox/internal/SqueezeBoxAudioSink.java

```diff
@@ -63,21 +69,26 @@ public String getLabel(Locale locale) {
     }
 
     @Override
-    public void process(AudioStream audioStream) throws UnsupportedAudioFormatException {
+    public void process(AudioStream audioStream)
+            throws UnsupportedAudioFormatException, UnsupportedAudioStreamException {
         AudioFormat format = audioStream.getFormat();
         if (!AudioFormat.WAV.isCompatible(format) && !AudioFormat.MP3.isCompatible(format)) {
             throw new UnsupportedAudioFormatException("Currently only MP3 and WAV formats are supported: ", format);
         }
 
         String url;
-        if (audioStream instanceof FixedLengthAudioStream) {
+        if (audioStream instanceof URLAudioStream) {
+            url = ((URLAudioStream) audioStream).getURL();
+        } else if (audioStream instanceof FixedLengthAudioStream) {
             // Since Squeezebox will make multiple requests for the stream, set a timeout on the stream
             url = audioHTTPServer.serve((FixedLengthAudioStream) audioStream, STREAM_TIMEOUT).toString();
 
             if (AudioFormat.WAV.isCompatible(format)) {
-                url += ".wav";
+                // TODO add FileAudioStream.EXTENSION_SEPARATOR
```

Comment:
@cweitkamp @martinvw Is there a reason you didn't add the extension separator as part of this change.  The squeezebox binding is currently broken because there's no longer a separator in the file name.